### PR TITLE
CLOUD-353 Simplify PVC info obtaining

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:master-broker-PR-41-34a7db1

File: dbaas/list.go [hunk 1]
```diff
@@ -50,30 +56,25 @@ func (p Cmd) Describe(app Deploy) (string, error) {
...
 
-	cr, err := p.getPodInfo(app.DataPodName(0))
+	pvcList := &MultiplePVCk8sOutput{}
+	pvcsJSON, err := p.runCmd("kubectl", "get", "pvc", fmt.Sprintf("--selector=app.kubernetes.io/instance=%s,app.kubernetes.io/managed-by=%s", app.Name(), app.OperatorName()), "-o", "json")
 	if err != nil {
 		return "", err
 	}
+	err = json.Unmarshal([]byte(pvcsJSON), &pvcList)
+	if err != nil {
+		return "", errors.Wrapf(err, "json prase")
```
Comment (dAdAbird) [inline dbaas/list.go]: s/prase/parse

File: dbaas/list.go [hunk 2]
```diff
@@ -50,30 +56,25 @@ func (p Cmd) Describe(app Deploy) (string, error) {
 		return "", errors.Wrapf(err, "json prase")
 	}
 
-	cr, err := p.getPodInfo(app.DataPodName(0))
+	pvcList := &MultiplePVCk8sOutput{}
+	pvcsJSON, err := p.runCmd("kubectl", "get", "pvc", fmt.Sprintf("--selector=app.kubernetes.io/instance=%s,app.kubernetes.io/managed-by=%s", app.Name(), app.OperatorName()), "-o", "json")
 	if err != nil {
 		return "", err
 	}
+	err = json.Unmarshal([]byte(pvcsJSON), &pvcList)
```
Comment (dAdAbird) [inline dbaas/list.go]: `pvcList` is a pointer already

File: dbaas/list.go [hunk 3]
```diff
@@ -50,30 +56,25 @@ func (p Cmd) Describe(app Deploy) (string, error) {
...
-				if err != nil {
-					return "", err
-				}
-				AllocatedStorage[app.PodTypes()[pod]] = string(qt)
-			}
+	for volume := range pvcList.Items {
+		PVCs[pvcList.Items[volume].Labels["app.kubernetes.io/component"]] = *pvcList.Items[volume].Spec.StorageClassName
+		qt, err := pvcList.Items[volume].Status.Capacity["storage"].MarshalJSON()
+		if err != nil {
+			return "", err
```
Comment (dAdAbird) [inline dbaas/list.go]: more error context needed

**Review by dAdAbird — APPROVED** [review]

File: dbaas/list.go (hunk 1 above)
Comment (cap1984) [reply dbaas/list.go]: done

File: dbaas/list.go (hunk 2 above)
Comment (cap1984) [reply dbaas/list.go]: done

File: dbaas/list.go (hunk 3 above)
Comment (cap1984) [reply dbaas/list.go]: done

**Review by delgod — APPROVED** [review]

**Pull request merged**